raiden/raiden_service.py

```diff
@@ -903,6 +905,20 @@ def _direct_or_mediated_transfer(self, token_address, amount, identifier, direct
                 direct_channel.partner_state.address,
             )
 
+        elif direct_transfer_amount and direct_transfer_amount[1] != amount:
+            raise ValueError('You must not retry a transfer in a given channel'
+                             'with an used identifier and different amount',
+                             direct_transfer_amount)
+
+        elif direct_transfer_amount:
+            direct_transfer, _ = direct_transfer_amount
+
+            # if this transfer was already seen, just send again, no new state change
```

Comment:
Hmm, ok, but why can't we send the message again? It could be useful in the case of a timeout, and would not pose any security risk, as the state change was already enforced, and it'd only retry it over user's request (users should have the option to retry a failed transfer, using the same identifier/amount as the previous request)

Reply:
It's retried, but it's not under the user control, that is up to the protocol to decide when a message is sent.

Reply:
Ok, I'll change the comment to state that.